src/lib.rs

```diff
@@ -8,6 +8,10 @@ use hal_core::{boot::BootInfo, mem, Architecture};
 
 use alloc::vec::Vec;
 
+mod wasm;
+
+const HELLOWORLD_WASM: &[u8] = include_bytes!("helloworld.wasm");
+
 pub fn kernel_main<A>(bootinfo: &impl BootInfo<Arch = A>) -> !
```

Comment:
this clippy lint fires a lot of false positives on tracing macros, we should just turn it off